pkg/utils/kubernetes/utils.go

```diff
@@ -0,0 +1,34 @@
+package kubernetes
+
+import (
+	"errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func CustomResourceDefinitionExists(gvk schema.GroupVersionKind, cfg ...*rest.Config) (bool, error) {
```

Comment:
hmm, interesting approach to making config optional. Definitely a usability upside, but also potentially confusing. Do you have, or can you share any references where this approach is discussed / evaluated?

Reply:
no specific reference, more like following what I was doing with Java's method overloading.  I just feel a func requires user to pass in nil is redundant, so we can make it optional.  I did see [an example](https://github.com/myeung18/console-operator/blob/04ecd9f1186c74b23bc599b0471b9e9678a4b759/test/e2e/framework/clientset.go#L36) related to this: that kubeconfig could be optional.  I am interested in your point of view, why it is potentially confusing, but not redundant?

Reply:
It is confusing because your input right now is a variadic parameter (basically an array). As a consumer of your API, that means I can use zero, one, or multiple configs. That's not true, of course, as you expect either zero or one. I understand that it makes for nice code, without having to provide a nil, but there is a clear downside.

Why would anyone need to provide a config? If only for unit testing, then we can have this function not take a config, but call another private function within the same class that takes the config. Unit tests would then be able to call that private function directly.

Reply:
I am putting that variadic param was meant to make an optional param, as Go doesn't seem to have other supported way to create an optional param. This variadic param does create confusion here. I will change it to accept *rest.Config.

I dont understand that private function call by test cases directly part.
Say I have this call, not called with a config.:
CustomResourceDefinitionExists(gvk) -> since no cfg given --> call a_private_getConfig_func. to get one. 

How the Unit tests calling that private func directly will help to test that CustomResourceDefinitionExists? or you mean just to test this private func?

Reply:
Right, the public function would just have a line to create a config and call the private function. The unit test would directly call and test the private function.